### PR TITLE
Add Snapshot DS (PowerScale 1.1)

### PR DESCRIPTION
**Pull request by doriac11**
# Description
- Added the PowerScale Snapshot Datasource.
(For the next release)

# ISSUE TYPE
- Feature Pull Request

##### RESOURCE OR DATASOURCE NAME
Snapshot Datasource

##### OUTPUT
image

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 80% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Unit tests
- [x] Acceptance tests

### CODE REVIEW

**Comment by doriac11** [comment]
run e2e test

**Comment by doriac11** [comment]
run e2e test

**Comment by doriac11** [comment]
run e2e test

File: powerscale/models/snapshot.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+type SnapshotDataSourceModel struct {
+	ID        types.String          `tfsdk:"id"`
+	Snapshots []SnapshotDetailModel `tfsdk:"snapshots_details"`
+	// filter
+	SnapshotFilter *SnapshotFilterType `tfsdk:"filter"`
+}
+
+// SnapshotFilterType describes the filter data model.
+type SnapshotFilterType struct {
+	Path types.String `tfsdk:"path"`
```
Comment (nikitajoshi1) [inline powerscale/models/snapshot.go:32]: @doriac11 Can you add additional filters which are supported by the API itself like other datasources?

File: powerscale/provider/snapshot_datasource.go [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+// Metadata describes the data source arguments.
+func (d *SnapshotDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_snapshot"
+}
+
+// Schema describes the data source arguments.
+func (d *SnapshotDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		// This description is used by the documentation generator and the language server.
+		MarkdownDescription: "Snapshots Datasource. This datasource is used to query the existing Snapshots from PowerScale array. The information fetched from this datasource can be used for getting the details / for further processing in resource block. PowerScale Snapshots is a logical pointer to data that is stored on a cluster at a specific point in time.",
```
Comment (nikitajoshi1) [inline powerscale/provider/snapshot_datasource.go:56]: @doriac11 Can you remove additional _Snapshots Datasource._ in the description to make it consistent with others as per the format?

File: powerscale/models/snapshot.go (hunk 1 above)
Comment (doriac11) [reply powerscale/models/snapshot.go:32]: @nikitajoshi1 , I read through those filters, but they didnt make much sense (to a user). I think for the most part users will want to filter on the snapshots related to a particular path.

File: powerscale/provider/snapshot_datasource.go (hunk 2 above)
Comment (doriac11) [reply powerscale/provider/snapshot_datasource.go:56]: Done!

**Comment by doriac11** [comment]
run e2e test

**Review by nikitajoshi1 — APPROVED** [review]

**Comment by doriac11** [comment]
run e2e test

**Review by shekhar-j — APPROVED** [review]

**Pull request merged**